diff --git a/cub/cmake/CubInstallRules.cmake b/cub/cmake/CubInstallRules.cmake
index f4cdb1514..a8b3b1940 100644
--- a/cub/cmake/CubInstallRules.cmake
+++ b/cub/cmake/CubInstallRules.cmake
@@ -12,7 +12,7 @@ install(DIRECTORY "${CUB_SOURCE_DIR}/cub"
 
 install(DIRECTORY "${CUB_SOURCE_DIR}/cub/cmake/"
   DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/cub"
-  REGEX .*header-search.cmake.* EXCLUDE
+  REGEX "(.*-header-search\.cmake|.*\.cmake\.in)" EXCLUDE
 )
 # Need to configure a file to store the infix specified in
 # CMAKE_INSTALL_INCLUDEDIR since it can be defined by the user
diff --git a/libcudacxx/cmake/libcudacxxInstallRules.cmake b/libcudacxx/cmake/libcudacxxInstallRules.cmake
index 7de8d5bb4..1c1ed5cb0 100644
--- a/libcudacxx/cmake/libcudacxxInstallRules.cmake
+++ b/libcudacxx/cmake/libcudacxxInstallRules.cmake
@@ -22,7 +22,7 @@ install(DIRECTORY "${libcudacxx_SOURCE_DIR}/include/nv"
 # Libcudacxx cmake package
 install(DIRECTORY "${libcudacxx_SOURCE_DIR}/lib/cmake/libcudacxx"
   DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake"
-  REGEX .*header-search.cmake.* EXCLUDE
+  REGEX "(.*-header-search\.cmake|.*\.cmake\.in)" EXCLUDE
 )
 
 # Need to configure a file to store CMAKE_INSTALL_INCLUDEDIR
